index.js

```diff
@@ -22,13 +20,57 @@ const checkCwdOption = options => {
 	}
 };
 
+function gitPatterns(cb, sync = true) {
```

Comment:
You should not mix sync and async. Use two different methods for async and sync.